staging/src/k8s.io/kube-aggregator/pkg/controllers/openapi/aggregator/downloader.go

```diff
@@ -21,32 +21,62 @@ import (
 	"fmt"
 	"net/http"
 	"strings"
+	"sync"
 
 	"k8s.io/apiserver/pkg/authentication/user"
 	"k8s.io/apiserver/pkg/endpoints/request"
-	"k8s.io/kube-openapi/pkg/cached"
 	"k8s.io/kube-openapi/pkg/validation/spec"
 )
 
+type CacheableDownloader interface {
+	UpdateHandler(http.Handler)
+	Get() (*spec.Swagger, string, error)
+}
+
 // cacheableDownloader is a downloader that will always return the data
 // and the etag.
 type cacheableDownloader struct {
+	name       string
 	downloader *Downloader
 	handler    http.Handler
 	etag       string
 	spec       *spec.Swagger
+
+	// mutex protects the handler which can be replaced
+	mutex sync.Mutex
 }
 
 // NewCacheableDownloader creates a downloader that also returns the etag, making it useful to use as a cached dependency.
-func NewCacheableDownloader(downloader *Downloader, handler http.Handler) cached.Value[*spec.Swagger] {
+func NewCacheableDownloader(apiServiceName string, downloader *Downloader, handler http.Handler) CacheableDownloader {
 	return &cacheableDownloader{
+		name:       apiServiceName,
 		downloader: downloader,
 		handler:    handler,
 	}
 }
+func (d *cacheableDownloader) UpdateHandler(handler http.Handler) {
+	d.mutex.Lock()
+	defer d.mutex.Unlock()
+	d.handler = handler
+}
+
+func (d *cacheableDownloader) getHandler() http.Handler {
```

Comment:
If you use an atomic handler, then I think this is mostly unneeded, which makes the whole function below simpler (i.e. getting the handler is nothing more than `handler.Load()`)